src/lib/auth.ts

```diff
@@ -417,53 +422,86 @@ export const parseUrl = (url: string) => {
   return parser;
 };
 
-export const createDemoUser = (classId: string, userType: UserType, userId: string, offeringId: string) => {
+export interface CreateFakeUserOptions {
+  appMode: AppMode;
+  classId: string;
+  userType: UserType;
+  userId: string;
+  offeringId: string;
+}
+
+export const createFakeUser = (options: CreateFakeUserOptions) => {
+  const {appMode, classId, userType, userId, offeringId} = options;
+  const className = `${appMode === "demo" ? "Demo" : "QA"} Class ${classId}`;
   if (userType === "student") {
     const student: StudentUser = {
       type: "student",
       id: userId,
-      portal: "demo",
+      portal: appMode,
       firstName: "Student",
       lastName: `${userId}`,
       fullName: `Student ${userId}`,
       initials: `S${userId}`,
-      className: `Demo Class ${classId}`,
-      classHash: `democlass${classId}`,
+      className,
+      classHash: `${appMode}class${classId}`,
       offeringId,
     };
     return student;
   }
   else {
-    userId = `${parseInt(userId, 10) + 1000}`;
+    const teacherUserId = `${parseInt(userId, 10) + 1000}`;
     const teacher: TeacherUser = {
       type: "teacher",
-      id: userId,
-      portal: "demo",
+      id: teacherUserId,
+      portal: appMode,
       firstName: "Teacher",
-      lastName: `${userId}`,
-      fullName: `Teacher ${userId}`,
-      initials: `S${userId}`,
-      className: `Demo Class ${classId}`,
-      classHash: `democlass${classId}`,
+      lastName: `${teacherUserId}`,
+      fullName: `Teacher ${teacherUserId}`,
```

Comment:
Not directly related to this PR, but I can't remember whether we've discussed before the need to store the `fullName` rather than computing it at run-time. Would a `fullName` accessor function serve the same purpose?

Reply:
Yes it would.

Reply:
Actually, that is just a plain object - the user model already stores the full name in the `name` object.